src/BatteryStats.cpp

```diff
@@ -830,9 +830,9 @@ void JbdBmsBatteryStats::updateFrom(JbdBms::DataPointContainer const& dp)
 }
 
 void VictronSmartShuntStats::updateFrom(VeDirectShuntController::data_t const& shuntData) {
-    BatteryStats::setVoltage(shuntData.batteryVoltage_V_mV / 1000.0, millis());
-    BatteryStats::setSoC(static_cast<float>(shuntData.SOC) / 10, 1/*precision*/, millis());
-    BatteryStats::setCurrent(static_cast<float>(shuntData.batteryCurrent_I_mA) / 1000, 2/*precision*/, millis());
+    BatteryStats::setVoltage(shuntData.batteryVoltage_V_mV / 1000.0, VeDirectShunt.getLastUpdate());
+    BatteryStats::setSoC(static_cast<float>(shuntData.SOC) / 10, 1/*precision*/, VeDirectShunt.getLastUpdate());
+    BatteryStats::setCurrent(static_cast<float>(shuntData.batteryCurrent_I_mA) / 1000, 2/*precision*/, VeDirectShunt.getLastUpdate());
```

Comment:
What about moving `_lastUpdate = VeDirectShunt.getLastUpdate();` from line 857 above those three methods and then use `_lastUpdate` instead of `VeDirectShunt.getLastUpdate()` ?

```suggestion
    _lastUpdate = VeDirectShunt.getLastUpdate()
    BatteryStats::setVoltage(shuntData.batteryVoltage_V_mV / 1000.0, _lastUpdate);
    BatteryStats::setSoC(static_cast<float>(shuntData.SOC) / 10, 1/*precision*/, _lastUpdate);
    BatteryStats::setCurrent(static_cast<float>(shuntData.batteryCurrent_I_mA) / 1000, 2/*precision*/, _lastUpdate);
```